SearchAndSort/EmployeeData.java

```diff
@@ -0,0 +1,510 @@
+package SearchAndSort;
+
+import java.util.PriorityQueue;
+import java.util.Map;
+import java.util.LinkedHashMap;
+import java.util.Comparator;
+import java.util.InputMismatchException;
+import java.io.IOException;
+import java.io.FileWriter;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.time.temporal.ChronoUnit;
+import java.util.Scanner;
+import java.util.regex.Pattern;
+
+public class EmployeeData {
+
+    Map<Integer, Employee> hashEmployeeDetail;
+
+    // Employee Class
+    static class Employee {
+        private int employeeId;
+        private String employeeName;
+        private String employeeEmail;
+        private int employeeAge;
+        private LocalDate dateOfBirth;
+
+        // default Constructor
+        Employee() {
```

Comment:
Is there a need of creating a default constructor explicitly? If not then remove it.